frontend/model/contracts/group.js

```diff
@@ -297,6 +302,43 @@ sbp('chelonia/defineContract', {
       //       bound to the UI in some location.
       return getters.groupSettings.mincomeCurrency && currencies[getters.groupSettings.mincomeCurrency].displayWithCurrency
     },
+    getChatRooms (state, getters) {
+      const active = {}
+      const deleted = {}
+      for (const cID in getters.currentGroupState.chatRooms) {
+        if (!getters.currentGroupState.chatRooms[cID].deletedDate) {
+          active[cID] = getters.currentGroupState.chatRooms[cID]
+        } else {
+          deleted[cID] = getters.currentGroupState.chatRooms[cID]
+        }
+      }
+      return { active, deleted, all: { ...active, ...deleted } }
+    },
+    getChatRoomIDsInSort (state, getters, rootState, rootGetters) {
```

Comment:
Contracts getters can never have access to `rootState` or `rootGetters`.

They only ever have access to their own state or getters.

The only place in a contract where it is safe to access the `rootState` or `rootGetters` is in a `sideEffect` - nowhere else.

Reply:
I used `rootState` and `rootGetters` in this group getter because those two are supported by vuex by default when we have several modules inside. I didn't parse gettersProxy if it is possible but anyway it worked well.
I could get the expected value of `rootGetters.isJoinedChatRoom` and made a expected list.
Regarding this getter `getChatRoomIdsInSort`, it shouldn't be here in group contract getters so I will replace it to the `state.js` file but I wanna say `rootGetters` is working inside individual contract getters.

Reply:
@Silver-IT 

> but I wanna say `rootGetters` is working inside individual contract getters.

If it is working it is only because it is used from a vue component, in which case this getter must be moved to `state.js` - it must not be here.

The contract getters are meant to be getters that work with contracts - and they are compatible with Vue.js as well.

This getter that you've written here will not work with contracts. So it doesn't belong in this file. It must be understood that contracts can only ever access their own state - never anything else. That is how the system is designed to work. So this getter - if it is used in a vue component, must be moved either to that vue component as a computed property, or into state.js.

The recommendation for getters that are only ever used in a single vue component is to include them as computed properties in that vue file. Whereas getters in state.js should be getters that are used in multiple vue files.

Reply:
> This getter that you've written here will not work with contracts. 

I know that the getters, we define inside the contract, can only access the state and getters of the contract. That is how gettersProxy designed. But why doesn't it work with contracts? Do you mean other contracts?

> The recommendation for getters that are only ever used in a single vue component is to include them as computed properties in that vue file. Whereas getters in state.js should be getters that are used in multiple vue files.

Noted. Thanks

Reply:
> I know that the getters, we define inside the contract, can only access the state and getters of the contract. That is how gettersProxy designed. But why doesn't it work with contracts? Do you mean other contracts?

@Silver-IT If contracts could access state outside of themselves - it would be impossible to keep them in sync between clients, because it is unpredictable what that outside state might be. Therefore contracts are not allowed to access any state that is outside of their own.

Reply:
Completely make sense.. I used this in the Vue component as I have used this only once in that component. :grin: